include/openssl/asn1.h

```diff
@@ -221,43 +221,41 @@ typedef struct ASN1_VALUE_st ASN1_VALUE;
 
 # define DECLARE_ASN1_FUNCTIONS_name(type, name) \
         DECLARE_ASN1_ALLOC_FUNCTIONS_name(type, name) \
-        DECLARE_ASN1_ENCODE_FUNCTIONS(type, name, name)
-
-# define DECLARE_ASN1_FUNCTIONS_fname(type, itname, name) \
```

Comment:
Please do not remove existing macros.  It's perfectly OK to make them aliases, like I imagine something like this:

``` C
# define DECLARE_ASN1_FUNCTIONS_fname(type, itname, name) \
         DECLARE_ASN1_FUNCTIONS_name(type, name)
```

(although I'm a little unsure what should happen with `itname`)

I also thinks it's OK to place these "disappearing" macros together somewhere in the header file with a comment saying something like `/* Legacy compatibility */`

Reply:
So I'll re-introduce `DECLARE_ASN1_FUNCTIONS_fname`, basically as it was before:
```
/* Legacy compatibility */
# define DECLARE_ASN1_FUNCTIONS_fname(type, itname, name) \
         DECLARE_ASN1_ALLOC_FUNCTIONS_name(type, name) \
         DECLARE_ASN1_ENCODE_FUNCTIONS(type, itname, name)
```
Keeping this macro will also require re-adding its support in `util/perl/OpenSSL/ParseC.pm`:
```
    { regexp   => qr/DECLARE_ASN1_FUNCTIONS_fname<<<\((.*),(.*),(.*)\)>>>/,
      massager => sub { return (<<"EOF");
int d2i_$3(void);
int i2d_$3(void);
int $3_free(void);
int $3_new(void);
DECLARE_ASN1_ITEM($2)
EOF
      }
    },
```
The one and only(!) use of `DECLARE_ASN1_FUNCTIONS_fname` was this:
```
DECLARE_ASN1_FUNCTIONS_fname(ASN1_TYPE, ASN1_ANY, ASN1_TYPE)
```
which I've simply expanded to:
```
DECLARE_ASN1_ALLOC_FUNCTIONS_name(ASN1_TYPE, ASN1_TYPE)
DECLARE_ASN1_ENCODE_FUNCTIONS(ASN1_TYPE, ASN1_ANY, ASN1_TYPE)
```
so I really doubt if it's worth keeping that macro alive, in particular there is already a  rather wild-grown zoo of such macros.

And how about all those (now entirely redundant) `_const` variants of auxiliary macros that I purged?
```
DECLARE_ASN1_FUNCTIONS_const
DECLARE_ASN1_ENCODE_FUNCTIONS_const
I2D_OF_const
ASN1_dup_of_const
ASN1_i2d_fp_of_const
ASN1_i2d_bio_of_const
IMPLEMENT_ASN1_FUNCTIONS_const
IMPLEMENT_ASN1_ENCODE_FUNCTIONS_const_fname
```
AFAICS they were there just as workarounds until this constification is achieved.
Keeping those would in my view just mean clutter, and in case anyone still uses them, he can straightforwardly update the use by stripping the `_const` part of the name or do a compatibility declaration.  
Yet if you prefer, I can add
```
#define DECLARE_ASN1_FUNCTIONS_const(type) \
        DECLARE_ASN1_FUNCTIONS(type)
```
etc. and, as far as needed, their support in `ParseC.pm`.

Reply:
The point is that this is a public header file, so there may be *others* using these macros. We don't want to cause unnecessary breakage.

Reply:
I see. So I've added compatibility decls for all the macros I mentioned above.
Since the decls are for _others_, who might (although these macros are not documented) use them elsewhere, I didn't restore support for them in `ParseC.pm`. Ok this way?